sklearn/cluster/dbscan_.py

```diff
@@ -125,6 +125,10 @@ def dbscan(X, eps=0.5, min_samples=5, metric='minkowski', metric_params=None,
         X_mask = X.data <= eps
         masked_indices = astype(X.indices, np.intp, copy=False)[X_mask]
         masked_indptr = np.cumsum(X_mask)[X.indptr[1:] - 1]
+
+        if X.indptr[0] == X.indptr[1] == 0:  # check if first row is all zero
+            masked_indptr[0] = 0
```

Comment:
`np.concatenate([0, np.cumsum(X_mask)][X.indptr[1:]]` looks like the best fix as it do not require any if condition and works for all cases. Thanks for your suggestion *@jnothman*.